examples/defaulter-gen/generators/defaulter.go

```diff
@@ -520,15 +520,16 @@ func populateDefaultValue(node *callNode, t *types.Type, tags string, commentLin
 	var defaultString string
 	if len(defaultMap) == 1 {
 		defaultString = defaultMap[0]
+	} else if len(defaultMap) > 1 {
+		klog.Fatalf("Found more than one default tag for %v", t.Kind)
 	}
 
-	t, depth := resolveTypeAndDepth(t)
-	if depth > 0 && defaultString == "" {
+	baseT, depth := resolveTypeAndDepth(t)
```

Comment:
Renamed `t->baseT` since `getNestedDefault` would always be invoked on the primitive base type like `string`, `int`, etc and always yield no nested default string. 

It seems the original intention of the code was to call `getNestedDefault` on the input argument. I've refactored `t->baseT` usages elsewhere where the base primitive element type was desired.